CMakeLists.txt

```diff
@@ -0,0 +1,356 @@
+cmake_minimum_required(VERSION 3.12)
+
+project(coap VERSION 4.3.0 LANGUAGES CXX C)
+
+set(LIBCOAP_API_VERSION 2)
+
+#
+# options to tweak the library
+#
+
+option(ENABLE_DTLS "Enable building with DTLS support" ON)
+option(WITH_OPENSSL "Use OpenSSL for DTLS functions" ON)
+option(WITH_GNUTLS "Use GnuTLS for DTLS functions" OFF)
+option(WITH_TINYDTLS "Use TinyDTLS for DTLS functions" OFF)
```

Comment:
TinyDTLS fails to work for me.

Reply:
I did not want to support tinyDTLS in the first place, waiting for upstream cmake support, otherwise we still need autoconf for windows

Reply:
we can print a fatal error if the user want TINYDTLS when building with cmake

Reply:
actually tinydtls is supported if its installed in the system though we are currently not building tinydtls from source e.g. submodule... this can be added in a following MR if still needed